src/inmanta/deploy/scheduler.py

```diff
@@ -384,7 +385,11 @@ async def report_resource_state(
                 if details.attributes.get("send_event", False):
                     provides: Set[ResourceIdStr] = self._state.requires.provides_view().get(resource, set())
                     if provides:
-                        self._work.deploy_with_context(provides, deploying=self._deploying)
+                        # Not sure about this priority. I believe this method is called when a resource has a new state
+                        # and hence, new version.
```

Comment:
Reminder to drop this comment.

As to the question: this method is called when a resource finishes deploy (or at least we only reach this branch of it in that case). The deploy that is triggered here is the propagation of events to its dependents. e.g. a service's resource deploys successfully -> we schedule the lsm state transfer resource for deploy.

I wonder if we might want a separate priority level for event propagation. I would rank it pretty high (urgent) for fast event response, but perhaps given how eagerly we currently send them out (because we'll still default to `receive_events=True`), that might not be the best idea at the moment.

Overall, perhaps we should simply schedule it with the same priority as the resource that just finished? We could pass in the information to this method, or we could just query the agent queues mapping. The first is more direct, but it feels a bit clunky, given how generic this method is, so I think I prefer the second.

Reply:
Hmm, we don't even keep that information at the moment. I need to give this some more thought. Preliminary suggestion: make `AgentQueues._in_progress` a mapping from `Task` to `PrioritizedTask` instead of the current set, then query that here.

Reply:
Or perhaps have the `queue_get` method return `PrioritizedTask` instead of `Task`, and pass on the task context to this method?

I'll get back to this. I have to consider what fits best.

Reply:
Have you thought more about this case?

Reply:
I would have liked to just slam a hardcoded priority on this. But I'm not sure we can:
1. if we pick a high priority (low number), we essentially bump 80% of deploys to that priority, because event propagation is so eager at the moment. i.e. if `std::File("hello/world")` depends on `std::Directory("hello")`, then the file would get the high priority, even though there is no need for it at all.
2. if we pick a low priority (high number), we might starve actually relevant events (e.g. lsm) after repairs, facts, ... on the same agent.

As I see it, 1 is out of the question. 2 could be argued to be acceptable if real event listeners are expected to be on a separate agent. But iirc this is not the case for policy install, which is one of the two known event listeners (the other being lsm).

@wouterdb do you agree that this invalidates our conclusion? Mainly 1: I initially posed that we would bump the priority on event listeners, but I was overlooking that we would actually bump it on non-listeners as well (because `receive_events=True` by default), which is the core of the problem.

So, if Wouter agrees, how do we implement it? My preference is to put it in `AgentQueues._in_progress` as `Mapping[Task, TaskPriority]`, mainly because this allows us to easily bump in-progress tasks' priority as well, in the `deploy_with_context` path. By bumping in-progress tasks' priorities, we essentially bump their events as well. Wdyt?